slither/utils/type.py

```diff
@@ -32,24 +59,52 @@ def export_nested_types_from_variable(variable):
     return l
 
 
-def export_return_type_from_variable(variable):
+def export_return_type_from_variable(variable_or_type: Union[Type, Variable], all_types: bool = True) -> List[Type]:
     """
-    Return the type returned by a variable
-    :param variable
+    Return the type returned by a variable.
+    If all_types set to false, filter array/mapping. This is useful as the mapping/array in a structure are not
+    returned by solidity
+
+    :param variable_or_type
+    :param all_types
     :return: Type
     """
-    if isinstance(variable, MappingType):
-        return export_return_type_from_variable(variable.type_to)
-
-    if isinstance(variable, ArrayType):
-        return variable.type
-
-    if isinstance(variable.type, MappingType):
-        return export_return_type_from_variable(variable.type.type_to)
-
-    if isinstance(variable.type, ArrayType):
-        return variable.type.type
-
-    return variable.type
-
-
+    from slither.core.declarations import Structure
+
+    if isinstance(variable_or_type, Type):
+        if isinstance(variable_or_type, MappingType):
+            if not all_types:
+                return []
+            return export_return_type_from_variable(variable_or_type.type_to)
+
+        if isinstance(variable_or_type, ArrayType):
+            if not all_types:
+                return []
+            return export_return_type_from_variable(variable_or_type.type)
+
+        if isinstance(variable_or_type, UserDefinedType) and isinstance(variable_or_type.type, Structure):
+            ret = []
+            for r in variable_or_type.type.elems_ordered:
+                ret += export_return_type_from_variable(r, all_types=False)
+
+            return ret
+
+        return [variable_or_type]
+    else:
+        if isinstance(variable_or_type.type, MappingType):
```

Comment:
Can we replace this with `export_return_type_from_variable(variable_or_type.type)`?